app/services/customers/update_invoice_grace_period_service.rb

```diff
@@ -4,27 +4,29 @@ module Customers
   class UpdateInvoiceGracePeriodService < BaseService
     def initialize(customer:, grace_period:)
       @customer = customer
-      @grace_period = grace_period
+      @grace_period = grace_period.to_i
       super
     end
 
     def call
-      if grace_period != customer.invoice_grace_period
+      old_grace_period = customer.invoice_grace_period.to_i
+      grace_period_diff = grace_period - old_grace_period
+
+      if grace_period != old_grace_period
         customer.invoice_grace_period = grace_period
         customer.save!
 
+        # NOTE: Update issuing_date on draft invoices.
+        customer.invoices.draft.each do |invoice|
+          invoice.issuing_date = invoice.issuing_date + grace_period_diff.days
```

Comment:
what happens if the issuing_date moves into the past here?

Reply:
Yeah, it's fine as long as we keep the `<=` in the `ready_to_be_finalized` scope, I think 👍